src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -0,0 +1,349 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lombok;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.RemoveAnnotationVisitor;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class LombokValueToRecord extends Recipe {
+
+    @Option(displayName = "useExactToString",
+            description = "When set the toString format from Lombok is used in the migrated record.")
+    boolean useExactToString;
+
+    @JsonCreator
+    public LombokValueToRecord(final @JsonProperty("useExactToString") boolean useExactToString) {
+        this.useExactToString = useExactToString;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Convert Value class to Record";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Convert Lombok Value annotated classes to standard Java Record classes in Java 17 or higher.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("lombok");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+    @Override
+    public boolean causesAnotherCycle() {
+        return true;
+    }
+
+    @Override
+    public int maxCycles() {
+        return 2;
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        final TreeVisitor<?, ExecutionContext> check = Preconditions.and(
+                new UsesJavaVersion<>(17)
```

Comment:
Perhaps good to limit recipe execution here early by checking for the presence of `lombok.Value` through something like 
```suggestion
        final TreeVisitor<?, ExecutionContext> check = Preconditions.and(
                new UsesJavaVersion<>(17),
                new UsesType("lombok.Value")
```
Reviewing from GitHub, so the above might still need an import for instance.

Reply:
I like the suggestion but since I need to run the recipe twice I am not sure if this would break things because from my understanding in the second run the lombok Value classes would be already adjusted and therefore the import would be missing.

I will look into it and adjust if possible.

Reply:
Addressed with the refactoring to ScanningRecipe